components/ws-manager-mk2/controllers/workspace_controller.go

```diff
@@ -205,24 +207,14 @@ func (r *WorkspaceReconciler) actOnStatus(ctx context.Context, workspace *worksp
 
 	// we've disposed already - try to remove the finalizer and call it a day
 	case workspace.Status.Phase == workspacev1.WorkspacePhaseStopped:
-		var foundFinalizer bool
-		n := 0
-		for _, x := range pod.Finalizers {
-			if x != gitpodPodFinalizerName {
-				pod.Finalizers[n] = x
-				n++
-			} else {
-				foundFinalizer = true
-			}
-		}
-		pod.Finalizers = pod.Finalizers[:n]
-		err := r.Client.Update(ctx, pod)
-		if err != nil {
-			return ctrl.Result{Requeue: true}, err
+		hadFinalizer := controllerutil.ContainsFinalizer(pod, gitpodPodFinalizerName)
+		controllerutil.RemoveFinalizer(pod, gitpodPodFinalizerName)
+		if err := r.Client.Update(ctx, pod); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to remove gitpod finalizer: %w", err)
```

Comment:
The original behavior, if there was an error updating, was to requeue:

```go
err := r.Client.Update(ctx, pod)
if err != nil {
    return ctrl.Result{Requeue: true}, err
}
```

How come we no longer requeue if there was an error?

Reply:
`Requeue: true` is implicit if an error is returned, we can actually remove the requeue in other places as well.

From the [docs](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile#Reconciler):
```
// The Controller will requeue the Request to be processed again if an error is non-nil or
// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
```

Reply:
I see, thanks for the heads up, @WVerlaek ! Marking as resolved.